x-pack/winlogbeat/module/sysmon/config/winlogbeat-sysmon.js

```diff
@@ -474,6 +508,24 @@ var sysmon = (function () {
         evt.Put("file.code_signature.valid", signatureStatus === "Valid");
     };
 
+    var setAdditionalFileFieldsFromPath = function (evt) {
+        var filePath = evt.Get("file.path");
+        if (!filePath) {
+            return;
+        }
+
+        evt.Put("file.name", path.basename(filePath));
+        evt.Put("file.directory", path.dirname(filePath));
+
+        // path returns extensions with a preceding ., e.g.: .tmp, .png
+        // according to ecs the expected format is without it, so we need to remove it.
+        var ext = path.extname(filePath);
```

Comment:
What if the path does not have an extension? In that case I don't think we want to set the `file.extension` field.

Reply:
That makes sense, will change it 👍

Reply:
Also added a new event to the `.evtx` to test this case